internal/interfaces/http/tdex_connect_handler.go

```diff
@@ -32,31 +35,47 @@ type tdexConnect struct {
 	walletUnlockerSvc application.WalletUnlockerService
 	macaroonBytes     []byte
 	certBytes         []byte
-	serverPort        string
 	macaroonPath      string
 	certPath          string
+	serverAddress     string
+	protocol          string
 }
 
 func NewTdexConnectService(
 	repoManager ports.RepoManager,
 	walletUnlockerSvc application.WalletUnlockerService,
-	macaroonPath, certPath, serverPort string,
-) TdexConnectService {
+	macaroonPath, certPath, serverPort, host, protocol string,
+) (TdexConnectService, error) {
 	macBytes := readFile(macaroonPath)
 	certBytes := readFile(certPath)
 
+	p := protocol
+	if len(certBytes) > 0 {
+		if protocol == "" || protocol == httpsProtocol {
+			p = httpsProtocol
+		} else if protocol == httpProtocol {
+			return nil, errors.New("http protocol invalid with cert provided")
+		}
+	}
+
+	h := host
+	if host == "" {
+		h = "localhost"
+	}
```

Comment:
`CONNECT_URL_HOST` should be used in `AuthHandler` to build the correct address instead of always using `t.serverAddress`. I think it's ok for this service to have a simple `serverPort` field instead `serverAddress`.

Reply:
ref to [this comment](https://github.com/tdex-network/tdex-daemon/pull/625#discussion_r954813407).